### ISSUE TITLE
Stats: loading is very slow or unresponsive in Firefox

### ISSUE THREAD

**Issue by designsimply** [Automattic/wp-calypso#5879]
Steps to reproduce:

1. Log in to a WordPress.com or Jetpack site that has a lot of stats (i.e. >1M views per month)
1. Go to https://wordpress.com/ > My Sites > Stats > Months > [previous month] > Posts & Pages
1. Wait to see if the stats load

<strong>Result:</strong> the stats load very slowly in Chrome 50.0.2661.102 and an unresponsive script error appears every time in Firefox 46.0.1

> Warning: Unresponsive script
> 
> A script on this page may be busy, or it may have stopped responding. You can stop the script now, open the script in the debugger, or let the script continue.
> 
> Script: https://wordpress.com/calypso/…ts.ab876c7d5ecefd7be04a.min.js:8

<img width="472" alt="screen shot 2016-06-08 at wed jun 8 3 23 31 pm" src="https://cloud.githubusercontent.com/assets/1119271/15883612/15a9f6c0-2d8d-11e6-8847-dfbdbc19d8d5.png">

Reported in 2687524-t & 741360-f

/ht @danjjohnson @rachelsquirrel for the original reports

**Comment by blowery** [comment]
#5853 may help with this

**Comment by blowery** [comment]
Looks like it's not related. Tried to capture a profile of pulling up a bulky stats page and it's doing so much that the profiler gets stuck "recording" and the devtools hang. :)

**Comment by danjjohnson** [comment]
User in 2687524-t is having the same problem with weekly and daily stats (still very high numbers). He's using Chrome, and it works occasionally, but mostly fails.

**Comment by timmyc** [comment]
In the context of stats, we could very well get away with not running the entire post through the normalizer, and just work against the title since that is all that is shown here.  If it is the normalizer, wouldn't similar performance issues be happening on post/page listings as well?  I realized they are smaller datasets but seems like it could creep up there too.

**Comment by timmyc** [comment]
Well scratch that idea partially as `PostPerformance` uses `QueryPosts`.  @aduth probably helped alleviate some of the issue on the insights page in #5876 by reducing the number of posts queried to `1` vs the default of [20](https://github.com/Automattic/wp-calypso/blob/master/client/state/posts/constants.js#L5)

**Comment by blowery** [comment]
> In the context of stats, we could very well get away with not running the entire post through the normalizer, and just work against the title since that is all that is shown here.

Is this code flowing through the normalizer at all? I can't seem to find where the load happens.

**Comment by timmyc** [comment]
In the post performance module, the only decoding is happening [here](https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/stats/post-performance/index.jsx#L96).

**Comment by kriskarkoski** [comment]
More reports: 

#2682535-t 
#742057-f
#742276-f

All are using Firefox

**Comment by ashercantrell** [comment]
Seeing something similar in #394740-chat, but this user doesn't have a lot of stats and is seeing the error on wordpress.com/settings/general and wordpress.com/pages. Also, the JS file referenced is slightly different. It ends with b8085d7e5d702c41155c.min.js (the rest was truncated)

**Comment by timmyc** [comment]
I can re-create the issue on Windows + Firefox.  Still investigating, but it appears the timeout is happening in a date calculation.  Timeline wise - wondering if this might be related to the moving of i18n

**Comment by timmyc** [comment]
Tracing the unresponsive script to this call https://github.com/Automattic/i18n-calypso/blob/master/lib/mixin/index.js#L13 /cc @Tug

**Comment by blowery** [comment]
#5853 went out, FYI

**Comment by swoolz** [comment]
Waiting on user to confirm which browsers he used, but I think this may be a related report, too:

#2695885-t

**Comment by danjjohnson** [comment]
User in 2687524-t is still having the same problem with weekly and daily stats using Chrome. Is this likely to be a different issue? @Tug

**Comment by Tug** [comment]
It might be, I'm not sure as I didn't investigate the issue. @timmyc do you have an easy way to reproduce this?

**Comment by timmyc** [comment]
@danjjohnson I just logged in as the user and was able to pull up the stats for May.  It did take long due to the site being high traffic.  I performed the tests against my sandbox which is slower than the production API servers too.  The issue they are experiencing is not related to the original issue here.

VIP clients hit some of these issues too with trying to pull up stats for longer time periods.  Your suggestion of viewing Weekly/Daily are good work-arounds - or leveraging the API directly.
